src/SME.SGP.Aplicacao/CasosDeUso/CompensacaoAusencia/SalvarCompensasaoAusenciaUseCase.cs

```diff
@@ -99,6 +98,37 @@ public async Task Executar(long id, CompensacaoAusenciaDto compensacaoDto)
             }
         }
 
+        private async Task<PeriodoEscolarDto> BuscaPeriodo(Turma turma, int bimestre)
+        {
+            var tipoCalendario = await mediator.Send(new ObterTipoCalendarioPorAnoLetivoEModalidadeQuery(turma.AnoLetivo, turma.ModalidadeCodigo == Modalidade.EJA ? ModalidadeTipoCalendario.EJA : ModalidadeTipoCalendario.FundamentalMedio, turma.Semestre));
+
+            var parametroSistema = await mediator.Send(new ObterParametroSistemaPorTipoEAnoQuery(TipoParametroSistema.PermiteCompensacaoForaPeriodo, turma.AnoLetivo));
+
+            PeriodoEscolarDto periodo = null;
+            var consulta = await mediator.Send(new ObterPeriodoEscolarListaPorTipoCalendarioQuery(tipoCalendario.Id));
+            if (turma.ModalidadeCodigo == Modalidade.EJA)
+            {
+                if (turma.Semestre == 1)
+                    periodo = consulta.Periodos.FirstOrDefault(p => p.Bimestre == bimestre && p.PeriodoInicio < new DateTime(turma.AnoLetivo, 6, 1));
+                else
+                    periodo = consulta.Periodos.FirstOrDefault(p => p.Bimestre == bimestre && p.PeriodoFim > new DateTime(turma.AnoLetivo, 6, 1));
+            }
+            else
+                periodo = consulta.Periodos.FirstOrDefault(p => p.Bimestre == bimestre);
+
+            if (parametroSistema is not { Ativo: true })
+            {
+                var turmaEmPeriodoAberto =
+                    await mediator.Send(new TurmaEmPeriodoAbertoQuery(turma, DateTime.Today, bimestre, false,
+                        tipoCalendario.Id));
+
+                if (!turmaEmPeriodoAberto)
+                    throw new NegocioException($"Período do {bimestre}º Bimestre não está aberto.");
+            }
```

Comment:
Se o parâmetro pode bloquear o processo coloque a verificação dele antes de qualquer coisa, assim evita aquelas consultas de períodos de que não servirão se o processo for bloqueado aqui